src/jaybenne/sourcing.cpp

```diff
@@ -87,12 +101,38 @@ TaskStatus SourcePhotons(T *md, const Real t_start, const Real dt) {
               [[maybe_unused]] const Real &sbd = sb;
               [[maybe_unused]] const Real &vvd = vv;
               [[maybe_unused]] const Real &dtd = dt;
-              [[maybe_unused]] auto &opac = opacity;
+              [[maybe_unused]] auto *opac = opacity;
+              [[maybe_unused]] auto *mopac = mopacity;
+              [[maybe_unused]] const auto numind = numin;
+              [[maybe_unused]] const auto numaxd = numax;
+              [[maybe_unused]] const auto n_nubinsd = n_nubins;
               Real erad = 0.0;
               if constexpr (ST == SourceType::thermal) {
                 erad = (4.0 * sbd / vvd) * std::pow(temp, 4.0) * dv;
               } else if constexpr (ST == SourceType::emission) {
-                const Real emis = opac.Emissivity(rho, temp);
+                Real emis = -1.;
+                if constexpr (FT == FrequencyType::gray) {
+                  emis = opac->Emissivity(rho, temp);
+                } else if constexpr (FT == FrequencyType::multigroup) {
+                  // Construct emission CDF
+                  const Real dlnu = (std::log(numaxd) - std::log(numind)) / n_nubinsd;
+                  vmesh(b, fj::emission_cdf(0), k, j, i) = opac->EmissivityPerNu(
+                      rho, temp, std::exp(std::log(numind) + 0.5 * dlnu));
+                  for (int n = 1; n < n_nubinsd; n++) {
+                    const Real nu = std::exp(std::log(numind) + (n + 0.5) * dlnu);
+                    vmesh(b, fj::emission_cdf(n), k, j, i) =
+                        opac->EmissivityPerNu(rho, temp, nu) +
```

Comment:
Quick question on the units here - from the `singularity-opac` README, it looks like `EmissivityPerNu` returns units as energy density per time per frequency. Should there be a `dnu` multiplying `opac->EmissivityPerNu(rho, temp, nu)` here?